packages/flutter/test/widgets/fade_in_image_test.dart

```diff
@@ -22,21 +20,20 @@ class FadeInImageParts {
         assert(target != null);
 
   final ComponentElement fadeInImageElement;
-  final FadeInImageElements placeholder;
+  final FadeInImageElements? placeholder;
   final FadeInImageElements target;
 
-  State get state {
-    StatefulElement animatedFadeOutFadeInElement;
+  State? get state {
+    StatefulElement? animatedFadeOutFadeInElement;
     fadeInImageElement.visitChildren((Element child) {
       expect(animatedFadeOutFadeInElement, isNull);
       animatedFadeOutFadeInElement = child as StatefulElement;
     });
-    expect(animatedFadeOutFadeInElement, isNotNull);
-    return animatedFadeOutFadeInElement.state;
+    return animatedFadeOutFadeInElement?.state;
```

Comment:
should this be `!` instead of `?` since the previous expect seems to indicate that is is non-null here?